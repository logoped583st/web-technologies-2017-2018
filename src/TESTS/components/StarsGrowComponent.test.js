import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { HashRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import TabsStarsGrowReposComponent from '../../components/StarsGrowReposComponent';

configure({ adapter: new Adapter() });

describe('UserRepositories component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerUser: {
            login: 'logoped583st'
        },
        reducerRepos: [
            {
                name: 'adgfds',
                html_url: 'http://KAAAAKAA/',
                language: 'C#',
                private: true,
            },
            {
                name: 'logos',
                html_url: 'http://KAAAAKAA/',
                language: 'Java',
                private: false,
            },
        
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('UserRepositories component rendered succesfully', () => {
        const userRepositories = mount(
            <HashRouter>
                <TabsStarsGrowReposComponent store={store} />
            </HashRouter>
        );
        expect(mountToJson(userRepositories)).toMatchSnapshot();
    });

    
    it('UserRepositories rendered without user login', () => {
        const fakeInitialState = {
            reducerUser: {
                login: ''
            },
            reducerRepos: [
                {
                    name: 'asdasd',
                    html_url: 'http://KAAAAKAA/',
                    language: 'Java',
                    private: true,
                },

                
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userRepositories = mount(
            <HashRouter>
                <TabsStarsGrowReposComponent store={fakeStore} />
            </HashRouter>
        );
        expect(userRepositories.prop('login')).toBe(undefined);
        expect(userRepositories.prop('reducerRepos')).toBe(undefined);
        expect(mountToJson(userRepositories)).toMatchSnapshot();
    })
});