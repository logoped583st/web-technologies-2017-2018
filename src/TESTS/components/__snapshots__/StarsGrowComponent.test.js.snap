// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepositories component test UserRepositories component rendered succesfully 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(StarsGrowReposComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <StarsGrowReposComponent
        fetchGrowReposRequest={[Function]}
        fetchStarsReposRequest={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(StarsGrowReposComponent)>
  </Router>
</HashRouter>
`;

exports[`UserRepositories component test UserRepositories rendered without user login 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(StarsGrowReposComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <StarsGrowReposComponent
        fetchGrowReposRequest={[Function]}
        fetchStarsReposRequest={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(StarsGrowReposComponent)>
  </Router>
</HashRouter>
`;
